software/image_processor.py

```diff
@@ -121,10 +139,12 @@ def analyze_balls(self, t_balls, depth, fragments) -> list:
         return balls
 
     def analyze_baskets(self, t_basket, depth,  debug_color=(0, 255, 255)) -> list:
+        kernel = np.ones((3, 3), np.uint8)
```

Comment:
Declaring the kernel on every loop costs some performance, especially since it does not change over time.

Reply:
Moved to constructor.